client/src/pages/Home.js

```diff
@@ -1,30 +1,78 @@
 import MainTable from "../Component/MainTable";
 import SMSModal from "../Component/SMSModal";
 import "./Home.css";
-import { useState } from "react";
-import UploadFile from "./UploadFile";
+import React, { useState, useEffect } from "react";
+import Alert from "react-bootstrap/Alert";
+
+
 
 export function Home() {
-	const [checkedCheckboxes, setCheckedCheckboxes] = useState([]);
+	const [selectedClients, setSelectedClients] = useState([]);
+	const [clientData, setClientData] = useState([]);
+	const [error, setError] = useState(null);
+	const [isLoading, setIsLoading] = useState(false);
+	const [success, setSuccess] = useState(false);
+	const [danger, setDanger] = useState(false);
+
+
+	const loadClientData = () => {
+		setIsLoading(true);
+		fetch("/api/clients")
+			.then((response) => {
+				if (!response.ok) {
+					throw Error(
+						"There was an error getting the customer data, Please try again"
+					);
+				}
+				return response.json();
+			})
+			.then((data) => {
+				setClientData(data);
+				setError(null);
+				setSelectedClients([]);
+			})
+			.catch((err) => {
+				setError(err.message);
+			})
+			.finally(() => {
+				setIsLoading(false);
+			});
+	};
+
+	useEffect(() => {
+		loadClientData();
+	}, []);
 
 	const onHandleSelectedUserState = (isSelected, clientId ) => {
 		if (isSelected) {
-			setCheckedCheckboxes((previousCheckedCheckboxes) => {
-				return previousCheckedCheckboxes.concat(clientId);
+			setSelectedClients((previousSelectedClients) => {
+				return previousSelectedClients.concat(clientId);
 			});
 		} else {
-			setCheckedCheckboxes((previousCheckedCheckboxes) => {
-				return previousCheckedCheckboxes.filter((id) => clientId !== id);
+			setSelectedClients((previousSelectedClients) => {
+				return previousSelectedClients.filter((id) => clientId !== id);
 			});
 		}
 	};
 
+	const onSMSSent = () =>  {
+		loadClientData();
+		setSuccess(true);
+	};
+
+	const onSMSFailed = () => {
+		loadClientData();
+		setDanger(true);
+	};
 
 	return (
 		<main role="main">
-			<UploadFile />
-			<MainTable checkedCheckboxes={checkedCheckboxes} onHandleSelectedUserState={onHandleSelectedUserState} />
-			<SMSModal checkedCheckboxes={checkedCheckboxes} />
+			{success &&
+				<Alert variant="success" onClose={() => setSuccess(false)} dismissible> Message sent!</Alert>}
+			{danger &&
+				<Alert variant="danger" onClose={() => setDanger(false)} dismissible> Something went wrong! </Alert>}
```

Comment:
Rather than write `Alert` twice in the template, we could have one `Alert` and instead have a state variable `alert` with the structure `{ variant, message }` which can be set accordingly in `onSMSSent`, `onSMSFailed`, and the catch block of the fetch. It's okay for now so you can try doing the refactor if you feel like it!